### PR TITLE
bootstrap: add more unit tests

### PR DESCRIPTION
**Pull request by onur-ozkan**
self-explanatory

### CODE REVIEW

File: src/bootstrap/src/utils/change_tracker.rs [hunk 1]
```diff
@@ -2,6 +2,10 @@
 //! with the goal of keeping developers synchronized with important modifications in
 //! the bootstrap.
 
+#[cfg(test)]
+#[path = "../tests/change_tracker.rs"]
```
Comment (Mark-Simulacrum) [inline src/bootstrap/src/utils/change_tracker.rs:6]: Why are we doing this non-standard test movement? Just put the file in src/bootstrap/src/utils/change_tracker/test.rs -- that is the canonical way to add unit tests.

Comment (onur-ozkan) [reply src/bootstrap/src/utils/change_tracker.rs:6]: I followed the current approach in other bootstrap tests, but I agree. Will update it along with others.

**Comment by matthiaskrgr** [comment]
@bors r-  rollup=iffy

 failed in  https://github.com/rust-lang/rust/pull/120429#issuecomment-1913352683

**Pull request merged**